Defend The Divine/Assets/Scripts/GroupSpawner.cs

```diff
@@ -0,0 +1,105 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.VisualScripting;
+using UnityEditorInternal.Profiling.Memory.Experimental.FileFormat;
+using UnityEngine;
+
+/// <summary>
+/// GroupSpawner spawns groups of one enemy type numberOfGroups amount of 
+/// times, then destroys itself.
+/// Implemented this way to allow for more unique waves, e.g. wave 1 spawns a 
+/// group of 4 Enemy1 evry 5 seconds, while also spawning an Enemy2 every
+/// 7 seconds
+/// A "group" is a group of enemies spawned in unison, no matter the size 
+/// (Even if that size is 1)
+/// </summary>
+public class GroupSpawner : MonoBehaviour
+{
+    [SerializeField]
+    GameObject enemyType;
+    [SerializeField]
+    int groupSize;
+    [SerializeField]
+    int msBetweenGroups;
+    [SerializeField]
+    int numberOfGroups;
+
+    float groupSpawnTimestamp;
+    WaveManager waveManager;
+
+    /// <summary>
+    /// "Constructor" for GroupSpawner
+    /// </summary>
+    public void Initialize(
+        GameObject enemyType,
+        int groupSize,
+        int msBetweenGroups,
+        int numberOfGroups)
+    {
+        this.enemyType = enemyType;
+        this.groupSize = groupSize;
+        this.msBetweenGroups = msBetweenGroups;
+        this.numberOfGroups = numberOfGroups;
+
+        waveManager = GetComponent<WaveManager>();
```

Comment:
hmm, if this is called a lot, then it'd be calling GetComponent a decent amount which I think is fairly costly (not that important for our scale game, but still).
Can you get it from `waveManager = GameManager.Instance.WaveManager` instead?